src/ui/public/kuery/ast/kql.peg

```diff
@@ -1,123 +1,150 @@
 
-OrQuery
-  = left:AndQuery space 'or'i space right:OrQuery {
-    return addMeta(nodeTypes.function.buildNode('or', [left, right]), text(), location());
+OrList
+  = head:AndList Or tail:OrList {
+    return { or: [head, tail] };
   }
-  / AndQuery
+  / AndList
 
-AndQuery
-  = left:NotQuery space 'and'i space right:AndQuery {
-    return addMeta(nodeTypes.function.buildNode('and', [left, right]), text(), location());
+AndList
+  = head:NotList And tail:AndList{
+    return { and: [head, tail] };
   }
-  / NotQuery
+  / NotList
 
-NotQuery
-  = 'not'i space clause:Clause {
-    return addMeta(nodeTypes.function.buildNode('not', clause), text(), location());
+NotList
+  = Not list:SubList {
+    return { not: list };
   }
-  / Clause
+  / SubList
 
-Clause
-  = '(' subQuery:Query ')' {
-    return subQuery;
-  }
-  / Term
+SubList
+  = '(' Space* list:OrList Space* ')' { return list; }
+  / Literal
+
+Or
+  = Space+ 'or'i Space+
+
+And
+  = Space+ 'and'i Space+
+
+Not
+  = 'not'i Space+
 
-Term
-  = field:literal_arg_type space? ':' space? value:literal_arg_type {
-    return addMeta(nodeTypes.function.buildNodeWithArgumentNodes('is', [field, value]), text(), location());
+Literal
+  = SingleQuotedString
+  / DoubleQuotedString
+  / UnquotedLiteral
+
+SingleQuotedString
+  = "'" chars:(EscapedQuote / [^'])* "'" {
+    return chars.join('');
   }
-  / field:literal_arg_type space? ':' space? '[' space? gt:literal_arg_type space 'to'i space lt:literal_arg_type space? ']' {
-    return addMeta(nodeTypes.function.buildNodeWithArgumentNodes('range', [field, gt, lt]), text(), location());
+
+DoubleQuotedString
+  = '"' chars:(EscapedQuote / [^"])* '"' {
+    return chars.join('');
   }
-  / !Keywords literal:literal_arg_type { return literal; }
-
-literal_arg_type
- = literal:literal {
-   var result = addMeta(nodeTypes.literal.buildNode(literal), text(), location());
-   return result;
- }
-
-Keywords
-  = 'or'i / 'and'i / 'not'i
-
- /* ----- Core types ----- */
-
-literal "literal"
-  =  '"' chars:dq_char* '"' { return chars.join(''); } // double quoted string
-  / "'" chars:sq_char* "'" { return chars.join(''); } // single quoted string
-  / 'true'  { return true; } // unquoted literals from here down
-  / 'false' { return false; }
-  / 'null'  { return null; }
-  / string:[^\[\]()"',:=\ \t]+ { // this also matches numbers via Number()
-   var result = string.join('');
-   // Sort of hacky, but PEG doesn't have backtracking so
-   // a number rule is hard to read, and performs worse
-   if (isNaN(Number(result))) return result;
-   return Number(result)
+
+UnquotedLiteral
+  = chars:UnquotedCharacter+ {
+    const sequence = chars.join('').trim();
+    if (sequence === 'null') return null;
+    if (sequence === 'true') return true;
+    if (sequence === 'false') return false;
+    const number = Number(sequence);
+    return isNaN(number) ? sequence : number;
   }
 
-space
- = [\ \t\r\n]+
+EscapedQuote
+  = '\\' quote:('"' / '"') { return quote; }
```

Comment:
I think there might be a typo here, both quotes are the same. Shouldn't one of them be a single quote?

Reply:
2d86135